packages/kbn-monaco/src/painless/worker/lib/autocomplete_utils.ts

```diff
@@ -62,8 +90,10 @@ export const isDeclaringField = (activeTyping: string): boolean => {
 /**
  * Static suggestions serve as a catch-all most of the time
  * However, there are a few situations where we do not want to show them and instead default to the built-in monaco (abc) autocomplete
- * 1. If the preceding word is a type, e.g., "boolean", we assume the user is declaring a variable name
+ * 1. If the preceding word is a primitive type, e.g., "boolean", we assume the user is declaring a variable name
  * 2. If the string contains a "dot" character, we assume the user is attempting to access a property that we do not have information for
+ * 3. If the user is defining a variable with a boolean type, e.g., "boolean myBoolean ="
+ * 4. If the user is defining a string
```

Comment:
This entire logic should be more sophisticated, but I think we will improve upon this once ANTLR is integrated and we are able to provide contextual autocomplete.